### PR TITLE
Update spring boot to 2.3.3.RELEASE

### PR DESCRIPTION
**Pull request by ao508**
This is part of an effort to synchronize the spring-boot and spring versions across our repositories.

Co-authored-by: Niklas Reimer @nr23730
Co-authored-by: Karthik Kalletla <kalletlakarthik@gmail.com>
Co-authored-by: Angelica Ochoa <15623749+ao508@users.noreply.github.com>
Signed-off-by: Angelica Ochoa <15623749+ao508@users.noreply.github.com>

Supersedes https://github.com/cBioPortal/session-service/pull/31 and https://github.com/cBioPortal/session-service/pull/26

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,128 +1,125 @@
...
-    <!-- inherit defaults from spring boot -->
-    <parent>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <name>Session Service</name>
+  <description>RESTful API to cBioPortal/cbioportal sessions in MongoDB.</description>
+  <url>https://github.com/cBioPortal/session-service/</url>
+  <groupId>org.cbioportal.session_service</groupId>
+  <artifactId>session_service</artifactId>
+  <version>0.2.0</version>
```
Comment (inodb) [inline pom.xml:9]: @ao508 I don't think we want to downgrade: maybe `0.4.0` is better?
```suggestion
  <version>0.4.0</version>
```

Comment (ao508) [reply pom.xml:9]: ah I didnt catch that when merging the changes from the other 2 open PRs - updating now

**Review by inodb — APPROVED** [review]
Thanks @ao508 @kalletlak and @nr23730!! LGTM

**Comment by ao508** [comment]
@kalletlak  @nr23730 Just wanted to let you know that I'll be starting the spring-boot version sync with the cbioportal next, followed by some of our other repos which depend on cbioportal modules. The mismatched spring-boot (and spring in some cases) has been a huge pain in upstream apps so I was happy to see you two were also working on resolving the mismatched versioning we have.

**Review by kalletlak — APPROVED** [review]

**Pull request merged**